pkg/webhook/webhook.go

```diff
@@ -486,6 +513,16 @@ func MutateHandler(w http.ResponseWriter, req *http.Request) {
 					glog.Infof("network '%s/%s' doesn't use custom resources, skipping...", n.Namespace, n.Name)
 				}
 			}
+
+			/* parse the net-attach-def annotations for node selector label and add it to the desiredNsMap */
+			if ns, exists := networkAttachmentDefinition.ObjectMeta.Annotations[nodeSelectorKey]; exists {
+				nsNameValue := strings.Split(ns, "=")
```

Comment:
is it possible to have multiple key-value pairs defined for annotation: k8s.v1.cni.cncf.io/nodeSelector?
e.g. k8s.v1.cni.cncf.io/nodeSelector: disk=ssd,cpu=xeon
I do not have experience with this annotation.
Where is the documentation describing this annotation is what I am trying to get at? I cannot find it.

If it is limited to one key=value pair, then we should be opinionated if len(nsNameValue > 2) and return a bad request error.

Reply:
@martinkennelly Thanks, I too thought about this. so far we've seen a need for having single node selector label for the NAD. for example, select a node in which master interface is present for macvlan cni type, select a node in which ovs bridge is present for ovs-cni type. I think we would go with a check to limit with one key=value pair and throw a bad request error if its more than 1. WDYT ?
@JanScheurich Please let us know your thoughts.

Reply:
yeah. I think a limit check is good to prevent weird errors if we aren't supporting multiple key/value pairs in this patch.

Reply:
done.